btrdbextras/dq.py

```diff
@@ -0,0 +1,409 @@
+        -------
+        str
+            Table showing which distillates each Stream has available
+        """
+        fmt = "html" if notebook else None
+        table = [["uuid", "collection", "name"] + KNOWN_DISTILLER_TYPES]
+        for stream in self._streams:
+            temp = [str(stream.uuid)[:8] + "...", stream.collection, stream.name]
+            for distiller in KNOWN_DISTILLER_TYPES:
+                try:
+                    _ = stream[distiller]
+                    temp.append(u'\u2713')
+                except KeyError:
+                    temp.append("x")
+            table.append(temp)
+        return tabulate(table, headers="firstrow", tablefmt=fmt)
+
+    def contains_any_event(self, start=None, end=None, depth=30):
+        """
+        Indicates whether this group of streams contains any data quality events
+
+        Parameters
+        ----------
+        start: (optional) datetime, datetime64, float, str
+            start time of period to search for events
+        end: (optional) datetime, datetime64, float, str
+            end time of period to search for events
+        depth: (optional) int
+            The precision of the window duration as a power of 2 in nanoseconds.
+            e.g 30 would make the window duration accurate to roughly 1 second
+        
+        Returns
+        -------
+        dict[str, bool]
+            Returns dict indicating whether or not each of the underlying streams
+            contain any event
+        """
+        return {
+            str(stream.uuid): stream.contains_any_event(start=start, end=end, depth=depth)
+            for stream in self._streams
+        }
+
+    def contains_event(self, distil_type, start=None, end=None, depth=30):
+        """
+        Indicates whether this group of streams contains a specific data quality event
+
+        Parameters
+        ----------
+        distil_type: str
+            The type of event to search for. Must be one of KNOWN_DISTILLER_TYPES
+        start: (optional) datetime, datetime64, float, str
+            start time of period to search for events
+        end: (optional) datetime, datetime64, float, str
+            end time of period to search for events
+        depth: (optional) int
+            The precision of the window duration as a power of 2 in nanoseconds.
+            e.g 30 would make the window duration accurate to roughly 1 second
+        
+        Returns
+        -------
+        dict[str, bool]
+            Returns dict indicating whether or each of the underlying streams contain
+            a certain event
+        """
+        out = {}
+        for stream in self._streams:
+            try:
+                contains = stream.contains_event(distil_type, start=start, end=end, depth=depth)
+            except KeyError:
+                # NOTE: this might be a bad idea. How to represent streams that do not have this 
+                # distillate stream?
+                contains = None
```

Comment:
Returning `None` in this case seems okay to me, though I'd expect `None` to return from `contains_event` (thus not needed the try/except).

Reply:
yea good call!